API.md

```diff
@@ -1,150 +1,117 @@
+- **`idpLogout`** - Boolean value to log the user out from the identity provider on application logout. Requires the issuer to provide a `end_session_endpoint` value. Default is `false`.
+- **`loginPath`** - Relative path to application login. Default is `/login`.
+- **`logoutPath`** - Relative path to application logout. Default is `/logout`.
+- **`redirectUriPath`** - Relative path to the application callback to process the response from the authorization server. This value is combined with the `baseUrl` and sent to the authorize endpoint as the `redirectUri` parameter. Default is `/callback`.
+- **`required`** - Use a boolean value to require authentication for all routes. Pass a function instead to base this value on the request. Default is `true`.
+- **`routes`** - Boolean value to automatically install the login and logout routes. See [the examples](EXAMPLES.md) for more information on how this key is used. Default is `true`.
 
--  `req.openid.user`: contains the user information, use this if you need display an attribute of the user. You can change what's end up here by using the `getUser` parameter of the `auth` middleware.
--  `req.openid.tokens`: is the instance of [TokenSet](https://github.com/panva/node-openid-client/blob/master/docs/README.md#tokenset).
--  `req.openid.client`: is an instance of te [OpenID Client](https://github.com/panva/node-openid-client/blob/master/docs/README.md#client).
--  `req.isAuthenticated()`: returns true if the request is authenticated.
+### Authorization Params Key
 
-If the request is not authenticated, `req.openid` is `undefined`.
+The `authorizationParams` key defines the URL parameters used when redirecting users to the authorization server to log in. If this key is not provided by your application, its default value will be:
 
-Every `res` object gets the following methods:
+```js
+{
+  response_type: "id_token",
+  response_mode: "form_post",
+  scope: "openid profile email"
+}
+```
 
--  `res.openid.login(params)`: trigger an authentication request from any route. It receives the following parameters:
-  -  `params.returnTo`: The url to return to after authentication. Defaults to the current url for GETs and `baseURL` for other methods.
-  -  `params.authorizationParams`: additional parameters for the authorization call.
--  `res.openid.logout(params)`: trigger the openid connect logout if supporter by the issuer.
-  -  `params.returnTo`: The url to return to after sign out. Defaults to the `baseURL` for other methods.
+A new object can be passed in to change what is returned from the authorization server depending on your specific scenario.
 
-## Authorization handling
+For example, to receieve an access token for an API, you could initialize like the sample below. Note that `response_mode` can be omitted because the OAuth2 default mode of `query` is fine:
```

Comment:
Spelling:

```suggestion
For example, to receive an access token for an API, you could initialize like the sample below. Note that `response_mode` can be omitted because the OAuth2 default mode of `query` is fine:
```